bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/SearchResultServlet.java

```diff
@@ -16,282 +16,291 @@
+        "sling.servlet.resourceTypes=cq/Page",
+        "sling.servlet.extensions=json",
+        "sling.servlet.methods=GET"
+    }
 )
-public class SearchResultServlet extends SlingSafeMethodsServlet {
+public final class SearchResultServlet extends SlingSafeMethodsServlet {
 
+    /**
+     * Selector to trigger the search servlet.
+     */
     protected static final String DEFAULT_SELECTOR = "searchresults";
+
+    /**
+     * Name of the query parameter containing the user query.
+     */
     protected static final String PARAM_FULLTEXT = "fulltext";
 
-    private static final String PARAM_RESULTS_OFFSET = "resultsOffset";
-    private static final String PREDICATE_FULLTEXT = "fulltext";
-    private static final String PREDICATE_TYPE = "type";
-    private static final String PREDICATE_PATH = "path";
-    private static final String NN_STRUCTURE = "structure";
+    /**
+     * Name of the query parameter indicating the search result offset.
+     */
+    protected static final String PARAM_RESULTS_OFFSET = "resultsOffset";
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(SearchResultServlet.class);
+    /**
+     * Name of the template structure node.
+     */
+    private static final String NN_STRUCTURE = "structure";
 
+    /**
+     * Query builder service.
+     */
     @Reference
     private transient QueryBuilder queryBuilder;
 
+    /**
+     * Language manager service.
+     */
     @Reference
     private transient LanguageManager languageManager;
 
+    /**
+     * Relationship manager service.
+     */
     @Reference
     private transient LiveRelationshipManager relationshipManager;
 
+    /**
+     * Model factory service.
+     */
+    @Reference
+    private transient ModelFactory modelFactory;
+
+    /**
+     * Bundle context.
+     */
+    private BundleContext bundleContext;
+
+    /**
+     * Activate the service.
+     *
+     * @param bundleContext The bundle context.
+     */
+    @Activate
+    protected void activate(final BundleContext bundleContext) {
+        this.bundleContext = bundleContext;
+    }
+
     @Override
-    protected void doGet(@NotNull SlingHttpServletRequest request, @NotNull SlingHttpServletResponse response)
-            throws IOException {
-        Page currentPage = getCurrentPage(request);
+    protected void doGet(@NotNull final SlingHttpServletRequest request, @NotNull final SlingHttpServletResponse response)
+        throws IOException {
+        Page currentPage = Optional.ofNullable(request.getResourceResolver().adaptTo(PageManager.class))
+            .map(pm -> pm.getContainingPage(request.getResource()))
+            .orElse(null);
         if (currentPage != null) {
-            Resource searchResource = getSearchContentResource(request, currentPage);
-            List<ListItem> results = getResults(request, searchResource, currentPage);
-            writeJson(results, response);
-        }
-    }
+            SlingBindings bindings = new SlingBindings();
```

Comment:
why are sling bindings introduced? Where are they used?

Reply:
I will double check, but I believe the sling bindings was related to fetching the Search component model.